### PR TITLE
[TaegisXDR] Major version update

### PR DESCRIPTION
**Pull request by rreed42**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
This update includes:

* Added support for "Fetch Incidents"
* Added support for bypassing certificate verification
* Replaced the "Service URL" input with "Taegis Environment"
* Added 2 new commands (taegis-execute-playbook and taegis-fetch-playbook-execution)
* Updated 2 existing commands (taegis-fetch-investigation-alerts and taegis-fetch-investigation)

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [x] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - This replaces the previous "Service URL" with "Environment" in the integration configuration which will break on upgrade. The previous "Service URL" was utilized prior to our release of multiple environments. Unfortunately, this meant that links back to the non-US1 environments did not function as there were no mappings to the other 2 environments. This change asks less of the customer/user while ensuring the correct API and portal URLs.
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by yucohen** [comment]
@rreed42 Looks great! Lets set up a quick demo

**Comment by rreed42** [comment]
> @rreed42 Looks great! Lets set up a quick demo

@yucohen Sure thing. Can you email me at rreed@secureworks.com and we can set up a quick time?

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py [hunk 1]
```diff
@@ -397,35 +628,42 @@ def main():
...
 
-        if command not in commands:
-            raise NotImplementedError(
-                f'The "{command}" command has not been implemented.'
-            )
-
         if command == "test-module":
             result = test_module(client)
             return return_results(result)
         else:
```
Comment (yucohen) [inline Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py:665]:
```suggestion
        elif command == "fetch-incidents":
            fetch_incidents(client, environment, max_fetch)
        else:
```

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py [hunk 2]
```diff
@@ -397,35 +628,42 @@ def main():
...
+        "taegis-execute-playbook": execute_playbook_command,
         "taegis-fetch-alerts": fetch_alerts_command,
         "taegis-fetch-investigation": fetch_investigation_command,
         "taegis-fetch-investigation-alerts": fetch_investigation_alerts_command,
+        "taegis-fetch-playbook-execution": fetch_playbook_execution_command,
         "taegis-update-investigation": update_investigation_command,
         "test-module": test_module,
     }
 
     PARAMS = demisto.params()
```
Comment (yucohen) [inline Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py:642]:
```suggestion
    PARAMS = demisto.params()
    max_fetch = PARAMS.get('max_fetch')
```

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py [hunk 3]
```diff
@@ -205,14 +262,126 @@ def fetch_alerts_command(client: Client, args=None):
     return results
 
 
-def fetch_investigation_alerts_command(client: Client, args=None):
+def fetch_incidents(client: Client, env: str, args=None):
```
Comment (yucohen) [inline Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py:265]:
```suggestion
def fetch_incidents(client: Client, env: str, page_size: int):
```

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py [hunk 4]
```diff
@@ -205,14 +262,126 @@ def fetch_alerts_command(client: Client, args=None):
...
+    """
+    Fetch Taegis Investigations for the use with "Fetch Incidents"
+    """
+    page_size = args.get("max_fetch", 200)
```
Comment (yucohen) [inline Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py:269]:
```suggestion
```

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py [hunk 5]
```diff
@@ -205,14 +262,126 @@ def fetch_alerts_command(client: Client, args=None):
...
+    now = datetime.now()
+    start_time = now - timedelta(days=1)  # Default start if first ever run
+    if last_run and "start_time" in last_run:
+        start_time = last_run.get("start_time")
+        variables["createdAfter"] = start_time
+
+    result = client.graphql_run(query=query, variables=variables)
+
+    if result.get("errors") and result["errors"]:
+        raise ValueError(f"Error when fetching investigations: {result['errors'][0]['message']}")
```
Comment (yucohen) [inline Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py:344]:
```suggestion
        raise DemistoException(f"Error when fetching investigations: {result['errors'][0]['message']}")
```

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py [hunk 6]
```diff
@@ -205,14 +262,126 @@ def fetch_alerts_command(client: Client, args=None):
...
+    }
+
+    last_run = demisto.getLastRun()
+    demisto.debug(f"Last Fetch Incident Run: {last_run}")
+
+    now = datetime.now()
+    start_time = now - timedelta(days=1)  # Default start if first ever run
+    if last_run and "start_time" in last_run:
+        start_time = last_run.get("start_time")
+        variables["createdAfter"] = start_time
```
Comment (yucohen) [inline Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py:339]:
```suggestion
    variables["createdAfter"] = start_time
```

Comment (yucohen) [reply Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py:339]: when we dont have the "last_run", then the start time should be this set (so we wont get older incidents)

**Review by yucohen — CHANGES_REQUESTED** [review]
@rreed42 Great work!
Please address the changes (dont forget to make sure the fetch works properly after) and than we can merge this

**Review by yucohen — APPROVED** [review]

**Pull request merged**